src/decorators/OrmConnection.ts

```diff
@@ -17,10 +18,20 @@ export function OrmConnection(connectionName: string = "default"): Function {
             return connectionManager.get(connectionName);
         };
 
+        let handler = <Handler> {
+            object: target,
+            value: getValue
+        };
+
         if (index !== undefined) {
-            Container.registerParamHandler({ type: target as Function, index: index, getValue: getValue });
-        } else {
-            Container.registerPropertyHandler({ target: target as Function /* todo: looks like typedi wrong type here */, key: propertyName, getValue: getValue });
+            handler.index = index;
+        }
+
+        if (propertyName !== undefined) {
+            handler.propertyName = propertyName;
         }
+
+        Container.registerHandler(handler);
```

Comment:
This should be simplified to:

```ts
Container.registerHandler({
     object: target,
     propertyName: propertyName,
     index: index,
     value: () => {
            const connectionManager = Container.get(ConnectionManager);
            if (!connectionManager.has(connectionName))
                throw new Error(`Cannot get connection "${connectionName}" from the connection manager. ` +
                `Make sure you have created such connection. Also make sure you have called useContainer(Container) ` +
                `in your application before you established a connection and importing any entity.`);

            return connectionManager.get(connectionName);
        }
});
```

also, `target` can be renamed to `object` and it can be even:


```ts
Container.registerHandler({ object, propertyName, index, value: () => {
            const connectionManager = Container.get(ConnectionManager);
            if (!connectionManager.has(connectionName))
                throw new Error(`Cannot get connection "${connectionName}" from the connection manager. ` +
                `Make sure you have created such connection. Also make sure you have called useContainer(Container) ` +
                `in your application before you established a connection and importing any entity.`);

            return connectionManager.get(connectionName);
        }
});
```